### DISCUSSION TITLE
Multiarch docker image

### DISCUSSION THREAD

**Discussion by hkad98** [duckdb/duckdb#4781] [Ideas]
Could you please consider providing multiarch docker image. I found [some docker images](https://hub.docker.com/search?q=duckdb) on docker hub. Unfortunately, all these images are unofficial.

**Comment by Alex-Monahan** [comment]
Hello! 

What is your main use case that requires Docker? Which language client are you using? One of the design goals of DuckDB is to be very simple to install so that we can avoid having to require complicated tooling like Docker.

Thanks for trying out DuckDB!

**Reply by hkad98** [reply]
Hello,

I do not have any specific use case. I was browsing through the docker hub and was surprised by the number of unofficial docker images. So I think it's a pity that there is no official one. On the other, I have to agree with you that it will be best to keep installation as simple as possible.

**Reply by chenjian2664** [reply]
@Alex-Monahan Hi, I think there are lots of test frameworks(i.e. Trino connector tests) were integrated with Docker already, so it's worth to have the official docker image.

**Reply by yasinzaehringer-paradime** [reply]
Our use case would have been: just integrating the duckdb image into our platform (which uses both x64 and ARM) without needing to think multi-arch since it's abstracted away by docker itself.

**Reply by Mause** [reply]
> Our use case would have been: just integrating the duckdb image into our platform (which uses both x64 and ARM) without needing to think multi-arch since it's abstracted away by docker itself.

Which language binding are you hoping to use? I'm not sure I fully understand your use case, given DuckDB can't be accessed over TCP

**Comment by dragonsahead** [comment]
Hi @Alex-Monahan, sorry for bringing back this discussion. The issue about Docker is that if you don't publish an official image then there will be lots of people that will publish unofficial ones out there and you'll end up with users that got the image from a creepy place without even knowing what they got (yes, that happens in 2024). On top of that, there's many teams/companies that only know about k8s and container orchestrators (that's their bread and butter since k8s took over the world), so not having an "official" image will end up with lots of users building their own, or worse, pulling others images. I hope this helps :)

**Reply by dragonsahead** [reply]
One more thing: adding a container image as an artifact to be released with the other available alternatives is pretty easy, you just bundle the binary inside a container and that's it
